SCons/Node/Python.py

```diff
@@ -152,12 +152,10 @@ def get_contents(self):
         Get contents for signature calculations.
         :return: bytes
         """
-        text_contents = self.get_text_contents()
-        try:
-            return text_contents.encode()
-        except UnicodeDecodeError:
-            # Already encoded as python2 str are bytes
-            return text_contents
+        contents = str(self.value).encode('utf-8')
+        for kid in self.children(None):
+            contents = contents + kid.get_contents()
```

Comment:
Can you change this to contents = contents + kid.get_csig()? Otherwise the memory usage of Value targets gets out of control.

Reply:
I've made a few extra commits, first changing `Value.get_csig()` to use child csigs instead of content, and then making `Value.get_text_contents()` to just be a concatenation of child csigs prepended with the stringified `value` attribute like you suggest. The `Value.get_contents()` then just return an encoded `get_text_contents()` and `Value.get_csig()` just wraps `get_text_contents()` as well.

I'm not sure where the 'contents' of a `Value` node is actually used, but the second change failed these tests:
```
SCons/SConfTests.py
test/Configure/ConfigureDryRunError.py
test/Configure/VariantDir-SConscript.py
test/Configure/VariantDir.py
test/Configure/basic.py
test/Configure/cache-not-ok.py
test/Configure/cache-ok.py
test/Configure/config-h.py
test/Configure/custom-tests.py
test/Configure/issue-3469/issue-3469.py
test/Configure/option--config.py
test/DVIPDF/DVIPDFFLAGS.py
test/Dir/source.py
test/Value.py
test/explain/basic.py
test/option-n.py
test/question/Configure.py
test/sconsign/script/Configure.py
test/textfile/textfile.py
```

The first change passes existing tests (after they've been modified to expect bytes from `get_contents()` calls in the previous PR commits).

I am running tests with Python 3.8.2 on Ubuntu 20.04 in a virtual environment.